### PR TITLE
[back] chore: Document and refresh dependencies of the backend

### PR DESCRIPTION
**Pull request by lfaucon**
to-do:

- [x] Read the [release notes for Django 4.1](https://docs.djangoproject.com/en/4.1/releases/4.1/) and make sure we are not missing anything important: 
- [x] Document usage of `django-sql-utils` and `django-filter`
- [x] ~~Fix setup of client secrets in dev-env (2e406e2 is not enough. It looks like saving the hashed secret directly does not work and the hashed secret gets hashed a second time on save in the django admin)~~
- [x] Decide what to do about warning in tests:
```
tournesol/tests/utils/mock_now.py:25
  /backend/tournesol/tests/utils/mock_now.py:25: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
    DEFAULT_DATE = datetime(2020, 1, 1, tzinfo=timezone.utc)

tournesol/tests/test_exports.py::ExportTest::test_all_export_comparisons_sorted_by_username_and_weekdate
  /backend/tournesol/tests/test_exports.py:505: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
    with MockNow.Context(datetime(2023, 1, 16, tzinfo=timezone.utc)):

tournesol/tests/test_exports.py::ExportTest::test_all_export_comparisons_sorted_by_username_and_weekdate
  /backend/tournesol/tests/test_exports.py:510: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
    with MockNow.Context(datetime(2023, 1, 22, tzinfo=timezone.utc)):

twitterbot/tests/test_tournesolbot.py::TestTournesolBotTopContributor::test_tweet_top_contributor_graph
twitterbot/tests/test_tournesolbot.py::TestTournesolBotTopContributor::test_tweet_top_contributor_graph
  /usr/local/lib/python3.9/site-packages/tweepy/auth.py:120: DeprecationWarning: OAuthHandler is deprecated; use OAuth1UserHandler instead.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

- [x] ~~Find a solution to not break production with client secrets when deploying~~

### CODE REVIEW

**Comment by GresilleSiffle** [comment]
About the warnings from Django:

The `django.utils.timezone.utc` alias to `datetime.timezone.utc` is deprecated. According to the warning and the documentation we shoud use `datetime.timezone.utc` instead.

About the warnings from Tweepy:

According to the release not we should simply use `OAuth1UserHandler` instead of `OAuthHandler`.

**Comment by lfaucon** [comment]
@GresilleSiffle I chose to not upgrade `django-oauth-tookit` which had breaking changes that were not straightforward to fix.

**Comment by lfaucon** [comment]
Thanks for sharing the links to release notes @GresilleSiffle . I did not find anything noteworthy in them.

I decided to not upgrade `django-oauth-tookit` because it has breaking changes that are not trivial to fix for us. I recommend upgrading it by itself in a separate PR. 

I think this PR could be merged as it is now.

File: backend/vouch/tests/test_trust_algo.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def test_trust_algo(self):
             self.assertIsNone(user.trust_score)
 
         trust_algo()
-        users = list(User.objects.all())
+        users = list(User.objects.all().order_by('username'))
```
Comment (amatissart) [inline backend/vouch/tests/test_trust_algo.py:140]: The same change could be applied line 149 too.

**Review by amatissart — APPROVED** [review]

**Pull request merged**